ui/components/multichain/app-header/app-header-unlocked-content.tsx

```diff
@@ -120,9 +120,11 @@ export const AppHeaderUnlockedContent = ({
   const shortenedAddress =
     internalAccount &&
     shortenAddress(normalizeSafeAddress(internalAccount.address));
-  const accountName = isMultichainAccountsState2Enabled
-    ? selectedMultichainAccount.metadata.name
-    : internalAccount.metadata.name;
+  const accountName =
+    (isMultichainAccountsState2Enabled
+      ? (selectedMultichainAccount?.metadata.name ??
+        internalAccount?.metadata.name)
```

Comment:
Not sure if we should use `internalAccount?.metadata.name` since it can create confusion, maybe just an empty string?